src/proctored-exam-settings/ProctoredExamSettings.jsx

```diff
@@ -470,20 +495,43 @@ function ProctoredExamSettings({ courseId, intl }) {
   useEffect(
     () => {
       dispatch(fetchExamSettingsPending(courseId));
-      StudioApiService.getProctoredExamSettingsData(courseId)
+
+      Promise.all([
+        StudioApiService.getProctoredExamSettingsData(courseId),
+        ExamsApiService.isAvailable() ? ExamsApiService.getCourseExamConfiguration(courseId) : Promise.resolve(),
+        ExamsApiService.isAvailable() ? ExamsApiService.getAvailableProviders() : Promise.resolve(),
+      ])
         .then(
-          response => {
-            const proctoredExamSettings = response.data.proctored_exam_settings;
+          ([settingsResponse, examConfigResponse, ltiProvidersResponse]) => {
+            const proctoredExamSettings = settingsResponse.data.proctored_exam_settings;
             setLoaded(true);
             setLoading(false);
             setSubmissionInProgress(false);
-            setCourseStartDate(response.data.course_start_date);
+            setCourseStartDate(settingsResponse.data.course_start_date);
             setEnableProctoredExams(proctoredExamSettings.enable_proctored_exams);
             setAllowOptingOut(proctoredExamSettings.allow_proctoring_opt_out);
-            setProctoringProvider(proctoredExamSettings.proctoring_provider);
             const isProctortrack = proctoredExamSettings.proctoring_provider === 'proctortrack';
             setShowProctortrackEscalationEmail(isProctortrack);
-            setAvailableProctoringProviders(response.data.available_proctoring_providers);
+
+            // The list of providers returned by studio settings are the default behavior. If lti_external
+            // is available as an option display the list of LTI providers returned by the exam service.
+            // Setting 'lti_external' in studio indicates an LTI provider configured outside of edx-platform.
+            // This option is not directly selectable.
+            const proctoringProvidersStudio = settingsResponse.data.available_proctoring_providers;
+            const proctoringProvidersLti = ltiProvidersResponse?.data || [];
+            setAllowLtiProviders(proctoringProvidersStudio.includes('lti_external'));
+            setLtiProctoringProviders(proctoringProvidersLti);
+            // flatten provider objects and coalesce values to just the provider key
+            setAvailableProctoringProviders(
+              proctoringProvidersLti.reduce((res, p) => [...res, p.name], []).concat(
```

Comment:
Without knowing the shape of `proctoringProvidersLti`, it's hard to understand what this does given these variable names. Can you be more descriptive? Is `p` provider? Is `res` response?

Reply:
cleaned this up a bit, res was just the internal result list passed through each iteration on the reduce.